app/Helpers/Bank/Nordigen/Transformer/TransactionTransformer.php

```diff
@@ -0,0 +1,149 @@
+ */
+
+namespace App\Helpers\Bank\Nordigen\Transformer;
+
+use App\Helpers\Bank\BankRevenueInterface;
+use App\Models\BankIntegration;
+use App\Utils\Traits\AppSetup;
+use Illuminate\Support\Facades\Cache;
+use Log;
+
+/**
+{
+  "transactions": {
+    "booked": [
+      {
+        "transactionId": "string",
+        "debtorName": "string",
+        "debtorAccount": {
+          "iban": "string"
+        },
+        "transactionAmount": {
+          "currency": "string",
+          "amount": "328.18"
+        },
+        "bankTransactionCode": "string",
+        "bookingDate": "date",
+        "valueDate": "date",
+        "remittanceInformationUnstructured": "string"
+      },
+      {
+        "transactionId": "string",
+        "transactionAmount": {
+          "currency": "string",
+          "amount": "947.26"
+        },
+        "bankTransactionCode": "string",
+        "bookingDate": "date",
+        "valueDate": "date",
+        "remittanceInformationUnstructured": "string"
+      }
+    ],
+    "pending": [
+      {
+        "transactionAmount": {
+          "currency": "string",
+          "amount": "99.20"
+        },
+        "valueDate": "date",
+        "remittanceInformationUnstructured": "string"
+      }
+    ]
+  }
+}
+*/
+
+class TransactionTransformer implements BankRevenueInterface
+{
+    use AppSetup;
+
+    public function transform($transactionResponse)
+    {
+        $data = [];
+
+        if (!array_key_exists('transactions', $transactionResponse) || !array_key_exists('booked', $transactionResponse["transactions"]))
+            throw new \Exception('invalid dataset');
+
+        foreach ($transactionResponse["transactions"]["booked"] as $transaction) {
+            $data[] = $this->transformTransaction($transaction);
+        }
+
+        return $data;
+    }
+
+    public function transformTransaction($transaction)
+    {
+
+        if (!array_key_exists('transactionId', $transaction) || !array_key_exists('transactionAmount', $transaction))
+            throw new \Exception('invalid dataset');
+
+        // description could be in varios places
+        $description = '';
+        if (array_key_exists('remittanceInformationStructured', $transaction))
+            $description = $transaction["remittanceInformationStructured"];
+        else if (array_key_exists('remittanceInformationStructuredArray', $transaction))
+            $description = implode(' \r\n', $transaction["remittanceInformationStructuredArray"]);
+        else if (array_key_exists('remittanceInformationUnstructured', $transaction))
+            $description = $transaction["remittanceInformationUnstructured"];
+        else if (array_key_exists('remittanceInformationUnstructuredArray', $transaction))
+            $description = implode(' \r\n', $transaction["remittanceInformationUnstructuredArray"]);
+        else
+            Log::warning("Missing description for the following transaction: " . json_encode($transaction));
```

Comment:
@turbo124 when parsing descriptions I save line-breaks with "\r\n". the frontends does not support this content. Should I change something? "<br>" is also not supported by frontends (I also tried this).

Reply:
@paulwer 

I think \n should be valid markup for new lines.

Are you not seeing this correctly in the react or flutter apps?

Reply:
only tested flutter and \r\n does not do anything. give me some min, i will test only with \n

Reply:
tested it only in flutter and it is also not working with \n.
I migrated programmcode to use \n only for now

### Table: Description with \r\n
![image](https://github.com/invoiceninja/invoiceninja/assets/52678724/911fd1ba-0d5b-42f3-84e2-586b270dd4d9)

### Table: Description with \n
![image](https://github.com/invoiceninja/invoiceninja/assets/52678724/fa0d80cb-1744-4741-a501-85d2642b1b03)

### Edit Mode
![image](https://github.com/invoiceninja/invoiceninja/assets/52678724/ac17d9dc-b86d-4c7f-a14e-f540b4e6a7fc)

### Datebase
![image](https://github.com/invoiceninja/invoiceninja/assets/52678724/e48b85b7-79eb-4ce7-b55d-5229fe134fe4)